test/numbers.jl

```diff
@@ -2771,7 +2771,8 @@ let types = (Base.BitInteger_types..., BigInt, Bool,
     for S in types
         for op in (+, -)
             if S === Float16 # type instability here?
-                @test_broken @inferred Base.promote_op(op, S)
+                # broken, fixme then remove this branch
```

Comment:
yeah `@inferred` isn't a first-class member of the test system in terms of behavior or return types, it's a separate mostly-independent utility